python/lsst/ts/genericcamera/driver/alliedvisioncamera.py

```diff
@@ -0,0 +1,302 @@
+# This file is part of ts_genericcamera.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import concurrent.futures
+import functools
+
+import vimba
+import yaml
+
+from . import basecamera
+from .. import exposure
+from ..fits_header_items_generator import FitsHeaderItemsGenerator, FitsHeaderTemplate
+
+MILLISECONDS = 10**3
+"Allied Vision cameras have timeouts in milliseconds."
+
+MICROSECONDS = 10**6
+"Allied Vision camera have exposure times in microseconds."
+
+
+class AlliedVisionCamera(basecamera.BaseCamera):
+    def __init__(self, log=None):
+        super().__init__(log=log)
+
+        self.id = 0
+        self.camera = None
+        self.frame = None
+        self.exposure_time = None
+        self.is_live_exposure = False
+        self.liveview_use_autoexposure = True
+        self.normal_image_type = None
+        self.loop = asyncio.get_running_loop()
+        self.executor = concurrent.futures.ThreadPoolExecutor()
+
+        self.tags = FitsHeaderItemsGenerator().generate_fits_header_items(
+            FitsHeaderTemplate.STARTRACKER
+        )
+        self.log.debug(f"Tags: {self.tags}")
+
+    @staticmethod
+    def name():
+        """Set camera name."""
+        return "AlliedVision"
+
+    def initialise(self, config):
+        """Initialise the camera with the specified configuration file.
+
+        Parameters
+        ----------
+        config : str
+            The name of the configuration file to load."""
```

Comment:
I think the trailing `"""` should be in a new line, no? I see this in other docstrings as well.

Reply:
OK, I'll fix them, but our other driver files show the same things, so I was just following convention.

Reply:
Fixed them all.